### PR TITLE
Support PSD constraints

### PR DESCRIPTION
**Pull request by AKS1996**
still some fixes to go

### CODE REVIEW

File: src/MOI_wrapper.jl [hunk 1]
```diff
@@ -39,7 +39,8 @@ const SUPPORTED_VECTOR_SETS = Union{
     MOI.Zeros, 
     MOI.Nonpositives,
     MOI.Nonnegatives,
-    MOI.SecondOrderCone
+    MOI.SecondOrderCone,
+    MOI.PositiveSemidefiniteConeTriangle
```
Comment (matbesancon) [inline src/MOI_wrapper.jl:43]:
```suggestion
    MOI.PositiveSemidefiniteConeTriangle,
```

File: src/MOI_wrapper.jl [hunk 2]
```diff
@@ -457,45 +457,27 @@ end
...
-            c = [1.0; 0.0; 0.0]
+        for con in model.con_idx
+            func = MOI.get(model.optimizer, MOI.ConstraintFunction(), con)
+            set = MOI.get(model.optimizer, MOI.ConstraintSet(), con)
+            MOIU.load_constraint(model.optimizer.model.optimizer, CI{typeof(func), typeof(set)}(0), func, set)
         end
+        
+        # now SCS data shud be allocated
+        A = sparse(
+            model.optimizer.model.optimizer.data.I,
```
Comment (matbesancon) [inline src/MOI_wrapper.jl:471]: `model.optimizer.model.optimizer.data.I,`

these lines look specific to one unique solver right?

File: src/MOI_wrapper.jl [hunk 3]
```diff
@@ -457,45 +457,27 @@ end
...
-                0.0   0.0  -1.0;
-                -1.0  0.0   0.0;
-                0.0  -1.0   0.0;
-            ]
-            b = [1.0; -0.707107; 0.0; 0.0; 0.0]
-            c = [1.0; 0.0; 0.0]
+        for con in model.con_idx
+            func = MOI.get(model.optimizer, MOI.ConstraintFunction(), con)
+            set = MOI.get(model.optimizer, MOI.ConstraintSet(), con)
+            MOIU.load_constraint(model.optimizer.model.optimizer, CI{typeof(func), typeof(set)}(0), func, set)
```
Comment (matbesancon) [inline src/MOI_wrapper.jl:466]: why 0 here?

File: src/MOI_wrapper.jl (hunk 2 above)
Comment (AKS1996) [reply src/MOI_wrapper.jl:471]: yes. extracting the matrices for SCS.jl was easiest

Comment (AKS1996) [reply src/MOI_wrapper.jl:471]: actually this is  the same code used in SCS.jl's optimize method https://github.com/jump-dev/SCS.jl/blob/master/src/MOI_wrapper.jl#L420

File: src/MOI_wrapper.jl (hunk 1 above)
Comment (AKS1996) [reply src/MOI_wrapper.jl]: added

File: src/MOI_wrapper.jl (hunk 3 above)
Comment (AKS1996) [reply src/MOI_wrapper.jl:466]: I'm not sure why, but when we run MOI.optimize! on a SCS.Optimizer model, this method load_constraint is called with `ConstraintIndex` for all constraints set as 0

Comment (AKS1996) [reply src/MOI_wrapper.jl:466]: had a rough time understanding why it is so. but this "hack" works to populate matrices A,b,c correctly (matched against diffcp)

File: src/MOI_wrapper.jl [hunk 4]
```diff
@@ -152,7 +153,7 @@ end
 
 
 function MOI.optimize!(model::Optimizer)
-    MOI.optimize!(model.optimizer)
+    solution = MOI.optimize!(model.optimizer)
```
Comment (blegat) [inline src/MOI_wrapper.jl:156]: `optimize!` returns `nothing`

File: src/MOI_wrapper.jl (hunk 3 above)
Comment (blegat) [reply src/MOI_wrapper.jl:466]: I wouldn't use the allocate-load API, just use `add_constraint`, you can check whether the solver needs a caching optimizer with `supports_default_copy_to`

Comment (blegat) [reply src/MOI_wrapper.jl:466]: That way, you will use a cache for SCS but not for Mosek for instance

Comment (blegat) [reply src/MOI_wrapper.jl:466]: By the way, the constraint index should be the one returned by load_constraint

File: src/MOI_wrapper.jl (hunk 4 above)
Comment (AKS1996) [reply src/MOI_wrapper.jl:156]: for SCS.jl, it returns `MOISolution`. can query it to get slack and dual solution easily

Comment (AKS1996) [reply src/MOI_wrapper.jl:156]: added this only for SCS.jl. indeed other solvers dont return anything

**Pull request merged**